src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const joinOperator = separator === null ? 'null' : separator;
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result = this.length - 1 === i
+          ? result + ''
+          : result + '' + joinOperator;
+      } else {
+        result = this.length - 1 === i
+          ? result + this[i]
+          : result + this[i] + joinOperator;
```

Comment:
Like in my other review, when adding strings I suggest using template literals for the readability sake and to omit such situations:
```
3 * 5 => 15
"3" * "5" => 15
2 + " fish" => "2 fish"
"3" + "5" => "35"  // hmmm...
```